pkg/operator/controller/ingress/controller.go

```diff
@@ -412,19 +412,63 @@ func setDefaultPublishingStrategy(ic *operatorv1.IngressController, infraConfig
 				changed = true
 			}
 
-			// If the ProviderParameters field does not exist for spec or status,
-			// just propagate (or remove) ProviderParameters in its entirety
-			// (as long as GCP parameters are specified one way or the other).
-			if specLB.ProviderParameters == nil && statusLB.ProviderParameters != nil && statusLB.ProviderParameters.GCP != nil ||
-				specLB.ProviderParameters != nil && specLB.ProviderParameters.GCP != nil && statusLB.ProviderParameters == nil {
-				ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters = specLB.ProviderParameters
-				changed = true
+			var lbType operatorv1.LoadBalancerProviderType
+			if specLB.ProviderParameters != nil {
+				lbType = specLB.ProviderParameters.Type
 			}
-
-			if specLB.ProviderParameters != nil && statusLB.ProviderParameters != nil &&
-				specLB.ProviderParameters.GCP != statusLB.ProviderParameters.GCP {
-				ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.GCP = specLB.ProviderParameters.GCP
-				changed = true
+			switch lbType {
+			case operatorv1.AWSLoadBalancerProvider:
+				// Detect changes to AWS connection idle timeout.
+				var specIdleTimeout, statusIdleTimeout metav1.Duration
+				if specLB.ProviderParameters != nil && specLB.ProviderParameters.AWS != nil && specLB.ProviderParameters.AWS.ClassicLoadBalancerParameters != nil {
+					if v := specLB.ProviderParameters.AWS.ClassicLoadBalancerParameters.ConnectionIdleTimeout; v != nil {
+						specIdleTimeout = *v
+					}
+				}
+				if statusLB.ProviderParameters != nil && statusLB.ProviderParameters.AWS != nil && statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters != nil {
+					if v := statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters.ConnectionIdleTimeout; v != nil {
+						statusIdleTimeout = *v
+					}
+				}
+				if specIdleTimeout != statusIdleTimeout {
```

Comment:
Should this also do some small validation on the timeout value to make sure a good value doesn't get changed into a bad value? I know for durations, there is no validation in the api.

Reply:
Good point, we might as well check for invalid values here.  I've changed the check below from `!= 0` to `> 0`.  Let me know if you had further validations in mind.